src/openqdc/datasets/base.py

```diff
@@ -66,16 +71,43 @@ class BaseDataset(torch.utils.data.Dataset):
     __force_methods__ = []
     energy_target_names = []
     force_target_names = []
+    __isolated_atom_energies__ = []
 
-    energy_unit = "hartree"
+    __energy_unit__ = "hartree"
+    __distance_unit__ = "ang"
+    __forces_unit__ = "hartree/ang"
+    __fn_energy__ = lambda x: x
+    __fn_distance__ = lambda x: x
+    __fn_forces__ = lambda x: x
 
-    def __init__(self) -> None:
+    def __init__(
+        self, energy_unit: Optional[str] = None, distance_unit: Optional[str] = None, cache_dir: Optional[str] = None
+    ) -> None:
+        set_cache_dir(cache_dir)
         self.data = None
+        self._set_units(energy_unit, distance_unit)
         if not self.is_preprocessed():
             entries = self.read_raw_entries()
             res = self.collate_list(entries)
             self.save_preprocess(res)
         self.read_preprocess()
+        self.__isolated_atom_energies__ = (
+            [IsolatedAtomEnergyFactory.get(en_method) for en_method in self.__energy_methods__]
+            if self.__energy_methods__
+            else None
+        )
+
```

Comment:
Why did we change this? i can see that we wrapped it inside a function but we are not using it